<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Hae Kyung Im" />

<meta name="date" content="2020-06-03" />

<title>QGT-Columbia-analysis-plan</title>

<script src="site_libs/header-attrs-2.1/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">QGT-Columbia-HKI</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hakyimlab/QGT-Columbia-HKI">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">QGT-Columbia-analysis-plan</h1>
<h4 class="author">Hae Kyung Im</h4>
<h4 class="date">2020-06-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>QGT-Columbia-lab/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200603code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200603)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200603code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200603)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabQGTColumbiaHKItreebc785821bf9670bd5d0c5c8607dd99e70b188fe9targetblankbc78582a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/hakyimlab/QGT-Columbia-HKI/tree/bc785821bf9670bd5d0c5c8607dd99e70b188fe9" target="_blank">bc78582</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabQGTColumbiaHKItreebc785821bf9670bd5d0c5c8607dd99e70b188fe9targetblankbc78582a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/hakyimlab/QGT-Columbia-HKI/tree/bc785821bf9670bd5d0c5c8607dd99e70b188fe9" target="_blank">bc78582</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    extras/.DS_Store

Unstaged changes:
    Modified:   analysis/analysis_plan.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/analysis_plan.Rmd</code>) and HTML (<code>docs/analysis_plan.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/bc785821bf9670bd5d0c5c8607dd99e70b188fe9/analysis/analysis_plan.Rmd" target="_blank">bc78582</a>
</td>
<td>
meliao
</td>
<td>
2020-06-09
</td>
<td>
Improved method of reading predicted expression (Thanks Tyson)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/e5b04c3c3b4b4b324a4095000a9fc20cf82fdac8/analysis/analysis_plan.Rmd" target="_blank">e5b04c3</a>
</td>
<td>
HKI qgt test
</td>
<td>
2020-06-09
</td>
<td>
height to cad
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/ad4f80bc1faea576aa7d2ad695b623a1ce2717ac/analysis/analysis_plan.Rmd" target="_blank">ad4f80b</a>
</td>
<td>
HKI qgt test
</td>
<td>
2020-06-09
</td>
<td>
height to cad
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/db5a1ae9059b7858884a15a5fc934e395770777d/analysis/analysis_plan.Rmd" target="_blank">db5a1ae</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-09
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/db5a1ae9059b7858884a15a5fc934e395770777d/docs/analysis_plan.html" target="_blank">db5a1ae</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-09
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/83e040c83516467b5bddfcb9713d1acd2533e6d5/analysis/analysis_plan.Rmd" target="_blank">83e040c</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-09
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/3fc9ab839b767199c71536c7dafd64c60ce334d6/analysis/analysis_plan.Rmd" target="_blank">3fc9ab8</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/d4f7c78e36f9c760cc1f099461ab6fef3b69299b/analysis/analysis_plan.Rmd" target="_blank">d4f7c78</a>
</td>
<td>
meliao
</td>
<td>
2020-06-08
</td>
<td>
Fixed merge
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/ca165eba1c70c5a0215d2baf1b8708f8c61d9b63/analysis/analysis_plan.Rmd" target="_blank">ca165eb</a>
</td>
<td>
meliao
</td>
<td>
2020-06-08
</td>
<td>
Added to code directory and updated analysis plan
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/b7047c7a020cdee3feb1360331d834da7951e9b2/analysis/analysis_plan.Rmd" target="_blank">b7047c7</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
more comments
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/b7047c7a020cdee3feb1360331d834da7951e9b2/docs/analysis_plan.html" target="_blank">b7047c7</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
more comments
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/c9cedba014a5e937ed45885a53e74d07ee3e1704/analysis/analysis_plan.Rmd" target="_blank">c9cedba</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
added brief explanation to chunks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/f754512802211ed81459c431842d37e2244c092d/analysis/analysis_plan.Rmd" target="_blank">f754512</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
removed - in spredixcan folder name
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/f754512802211ed81459c431842d37e2244c092d/docs/analysis_plan.html" target="_blank">f754512</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
removed - in spredixcan folder name
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/0ae45864c88c3e3e12b01a405fb72091f40ef3e2/analysis/analysis_plan.Rmd" target="_blank">0ae4586</a>
</td>
<td>
meliao
</td>
<td>
2020-06-08
</td>
<td>
Committing before merging in master
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/0ae45864c88c3e3e12b01a405fb72091f40ef3e2/docs/analysis_plan.html" target="_blank">0ae4586</a>
</td>
<td>
meliao
</td>
<td>
2020-06-08
</td>
<td>
Committing before merging in master
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/3489f1b2177989cb87892e481964b2197237aff7/analysis/analysis_plan.Rmd" target="_blank">3489f1b</a>
</td>
<td>
meliao
</td>
<td>
2020-06-08
</td>
<td>
Unfinished plotting changes. Committing before merge
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/b52e06aaa77c689d4c3a95cb7917b22e418c766b/analysis/analysis_plan.Rmd" target="_blank">b52e06a</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/b52e06aaa77c689d4c3a95cb7917b22e418c766b/docs/analysis_plan.html" target="_blank">b52e06a</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-08
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/a2b9918760e8105c91a8a5ae96bb35040e5eac7c/analysis/analysis_plan.Rmd" target="_blank">a2b9918</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-07
</td>
<td>
testing, reducing size of genotype file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/a2b9918760e8105c91a8a5ae96bb35040e5eac7c/docs/analysis_plan.html" target="_blank">a2b9918</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-07
</td>
<td>
testing, reducing size of genotype file
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/2257a0ffb347a9a5cd1137e163c96800ed4ebba8/analysis/analysis_plan.Rmd" target="_blank">2257a0f</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-06
</td>
<td>
raw figure links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/2257a0ffb347a9a5cd1137e163c96800ed4ebba8/docs/analysis_plan.html" target="_blank">2257a0f</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-06
</td>
<td>
raw figure links
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/efbfceb36ada5f9e4569a78c67bdc0ebdbcd93fc/analysis/analysis_plan.Rmd" target="_blank">efbfceb</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-06
</td>
<td>
raw urls for figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/973fc2b1f05c4350fc5d00c1575685469a5d0f5f/analysis/analysis_plan.Rmd" target="_blank">973fc2b</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
prelim notes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/517e8f71185d466a8656f911053629bbef93ec9e/analysis/analysis_plan.Rmd" target="_blank">517e8f7</a>
</td>
<td>
meliao
</td>
<td>
2020-06-05
</td>
<td>
Merge branch ‘master’ of <a href="https://github.com/hakyimlab/QGT-Columbia-HKI" class="uri">https://github.com/hakyimlab/QGT-Columbia-HKI</a>
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/409558fbe1e042fb8b10a2fe34b83b30e888445a/analysis/analysis_plan.Rmd" target="_blank">409558f</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2020-06-05
</td>
<td>
updated some paths
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/409558fbe1e042fb8b10a2fe34b83b30e888445a/docs/analysis_plan.html" target="_blank">409558f</a>
</td>
<td>
Yanyu Liang
</td>
<td>
2020-06-05
</td>
<td>
updated some paths
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/0deee3c4ef98fdd2ee955d953a93ef4a8f1e25a3/analysis/analysis_plan.Rmd" target="_blank">0deee3c</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
added TWMR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/0deee3c4ef98fdd2ee955d953a93ef4a8f1e25a3/docs/analysis_plan.html" target="_blank">0deee3c</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
added TWMR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/064f6eeb8a042a0f96f0198269cd6dde92a1a1da/analysis/analysis_plan.Rmd" target="_blank">064f6ee</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
added figures and slides under extras
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/064f6eeb8a042a0f96f0198269cd6dde92a1a1da/docs/analysis_plan.html" target="_blank">064f6ee</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
added figures and slides under extras
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/1b70eb7f2c25e0ab2e2e0795189fbc829b815d69/analysis/analysis_plan.Rmd" target="_blank">1b70eb7</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
updated prerequisites
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/1b70eb7f2c25e0ab2e2e0795189fbc829b815d69/docs/analysis_plan.html" target="_blank">1b70eb7</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
updated prerequisites
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/339f40ce0446812e8b6063570927916c29a9cb50/analysis/analysis_plan.Rmd" target="_blank">339f40c</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
added plan
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/339f40ce0446812e8b6063570927916c29a9cb50/docs/analysis_plan.html" target="_blank">339f40c</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
added plan
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/a3aa03eabde07481472c1e7e9333298978627411/analysis/analysis_plan.Rmd" target="_blank">a3aa03e</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
prerequisites added
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/09f5daef68b4080823d8a3f054e4a8fc29f4a894/analysis/analysis_plan.Rmd" target="_blank">09f5dae</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
fastenloc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/ccb1167f935aefc2aaeea4e93fe1c1ea284c06d1/analysis/analysis_plan.Rmd" target="_blank">ccb1167</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
knit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/ccb1167f935aefc2aaeea4e93fe1c1ea284c06d1/docs/analysis_plan.html" target="_blank">ccb1167</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
knit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/d427aee02c72daa61f997704fe6db760b4c96519/analysis/analysis_plan.Rmd" target="_blank">d427aee</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
minor comment sort1
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/4ca65b9cb29fb00da7bb35996999f7f73734e0d5/analysis/analysis_plan.Rmd" target="_blank">4ca65b9</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-05
</td>
<td>
edits 2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/f59cd026fb14b7aa0811300ec41043b93c5ccd1b/analysis/analysis_plan.Rmd" target="_blank">f59cd02</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/f59cd026fb14b7aa0811300ec41043b93c5ccd1b/docs/analysis_plan.html" target="_blank">f59cd02</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/1ad820c25746389acfe7282d4c62bd8cb3d67fa7/analysis/analysis_plan.Rmd" target="_blank">1ad820c</a>
</td>
<td>
meliao
</td>
<td>
2020-06-04
</td>
<td>
Added shell source command
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/d65c5558a2c3d732af0ed398d1bc99908b68f331/analysis/analysis_plan.Rmd" target="_blank">d65c555</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
twmr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/d65c5558a2c3d732af0ed398d1bc99908b68f331/docs/analysis_plan.html" target="_blank">d65c555</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
twmr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/682b6e2ba0b24fef25b003b0b289f587ed746d47/docs/analysis_plan.html" target="_blank">682b6e2</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/d3502e82abb1d356884a07f11856dddfd8bb9851/analysis/analysis_plan.Rmd" target="_blank">d3502e8</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
wflow_publish(“analysis/analysis_plan.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/4420fc7d9c5352115276af310f718e9cc83bd788/analysis/analysis_plan.Rmd" target="_blank">4420fc7</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
wflow_rename(“analysis/predixcan_analysis.Rmd”, “analysis/analysis_plan.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/QGT-Columbia-HKI/4420fc7d9c5352115276af310f718e9cc83bd788/docs/analysis_plan.html" target="_blank">4420fc7</a>
</td>
<td>
Hae Kyung Im
</td>
<td>
2020-06-04
</td>
<td>
wflow_rename(“analysis/predixcan_analysis.Rmd”, “analysis/analysis_plan.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
download data and software <a href="https://uchicago.box.com/s/zhapf2zfxcpj7thvq4sjnqale3emleum">from Box</a>. This will have copies of all the software repositories and the models</li>
</ul>
<p>Linux is the operating system of choice to run bioinformatics software. Here are offering two options</p>
<ul>
<li>Option 1: full setup, recommended for the linux-savvy with full setup</li>
<li>Option 2: pre-installed RStudio in Google cloud, recommended for people less familiar with linux</li>
</ul>
<p>The latest version of the analysis plan markdown document that generated this page is on <a href="https://github.com/hakyimlab/QGT-Columbia-HKI/blob/master/analysis/analysis_plan.Rmd">github here</a> rendered <a href="https://hakyimlab.github.io/QGT-Columbia-HKI/analysis_plan.html">here as an html page</a></p>
</div>
<div id="option-1" class="section level1">
<h1>Option 1</h1>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
install anaconda/miniconda</li>
<li><input type="checkbox" disabled="" />
define imlabtools conda environment <a href="https://github.com/hakyimlab/MetaXcan/blob/master/README.md#example-conda-environment-setup">how to here</a>, which will install all the python modules needed for this analysis session</li>
<li><input type="checkbox" disabled="" checked="" />
download software (copies of the repos are already included in the course folder QCT-Columbia-HKI/repos/)<ul>
<li>download metaxcan repo</li>
<li>download torus repo</li>
<li>download fastenloc repo</li>
<li>download TMWR repo</li>
</ul></li>
<li><input type="checkbox" disabled="" checked="" />
download prediction models from predictdb.org (a few models are included in the course folder QCT-Columbia-HKI/repos/)</li>
<li><input type="checkbox" disabled="" />
install R/RStudio/tidyverse package</li>
<li><input type="checkbox" disabled="" />
(optional) install workflowr package in R</li>
<li><input type="checkbox" disabled="" />
git clone <a href="https://github.com/hakyimlab/QGT-Columbia-HKI.git" class="uri">https://github.com/hakyimlab/QGT-Columbia-HKI.git</a></li>
<li><input type="checkbox" disabled="" />
start Rstudio (if you installed workflowr, you can just open the QGT-Columbia-HKI.Rproj)</li>
</ul>
</div>
<div id="option-2" class="section level1">
<h1>Option 2</h1>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
claim your Rstudio server IP address ()</li>
<li><input type="checkbox" disabled="" />
connect to the Rstudio server using the url you claimed (<a href="http://xxx.xxx.xxx.xxx:8787" class="uri">http://xxx.xxx.xxx.xxx:8787</a>)</li>
</ul>
</div>
<div id="initial-notices" class="section level1">
<h1>Initial notices</h1>
<ul>
<li><p>We ask you to fully participate in the hands on activities and to encourage that we may ask you to share your screen for pedagogic purposes.</p></li>
<li><p>As you run the analysis and programs, we ask you to respond the questions in <a href="https://docs.google.com/spreadsheets/d/1PKVfywvTu1RZuDCyHV4xt_cCngzhmjzw60_dHzLbLrc/edit#gid=145770066">this document</a>. Find the tab with your name and fill out the questions as you go along.</p></li>
<li><p>If you have any concerns about this, please ask me or one of the TAs for assistance.</p></li>
</ul>
</div>
<div id="both-options" class="section level1">
<h1>Both options</h1>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
update the analysis document</li>
</ul>
<pre class="bash"><code>PRE=&quot;/home/student/&quot;
cd $PRE/lab/
git pull </code></pre>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
activate the the imlabtools environment</li>
</ul>
<pre class="bash"><code>conda activate imlabtools</code></pre>
<p>** Notice that the bash chunks need to be copy-pasted to the terminal, not performed within the chunk.</p>
<div id="summary-of-analysis-plan" class="section level2">
<h2>Summary of analysis plan</h2>
<ul>
<li>predict whole blood expression</li>
<li>check how well the prediction works with GEUVADIS expression data</li>
<li>run association between predicted expression and a simulated phenotype</li>
<li>calculate association between expression levels and coronary artery disease risk using s-predixcan</li>
<li>fine-map the coronary artery disease gwas results using torus</li>
<li>calculate colocalization probability using fastenloc</li>
<li>run transcriptome-wide mendelian randomization in one locus of interest</li>
</ul>
<pre class="r"><code>suppressPackageStartupMessages(library(tidyverse))</code></pre>
</div>
</div>
<div id="transcriptome-wide-association-methods" class="section level1">
<h1>Transcriptome-wide association methods</h1>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
define some variables to access the data more easily within the R session. Run the following r chunk</li>
</ul>
<pre class="r"><code>print(getwd())

lab=&quot;/home/student/lab&quot;
CODE=glue::glue(&quot;{lab}/code&quot;)
source(glue::glue(&quot;{CODE}/extra_functions.R&quot;))
#source(glue::glue(&quot;code/extra_functions.R&quot;))

PRE=&quot;/home/student/QGT-Columbia-HKI&quot;
MODEL=glue::glue(&quot;{PRE}/models&quot;)
DATA=glue::glue(&quot;{PRE}/data&quot;)
RESULTS=glue::glue(&quot;{PRE}/results&quot;)
METAXCAN=glue::glue(&quot;{PRE}/repos/MetaXcan-master/software&quot;)
FASTENLOC=glue::glue(&quot;{PRE}/repos/fastenloc-master&quot;)
TORUS=glue::glue(&quot;{PRE}/repos/torus-master&quot;)
TWMR=glue::glue(&quot;{PRE}/repos/TWMR-master&quot;)</code></pre>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
define some variables to access the data more easily in the terminal. Run the following bash chunk. You will need to copy and paste the following chunk in the terminal</li>
</ul>
<pre class="bash"><code>export PRE=&quot;/home/student/QGT-Columbia-HKI&quot;
export LAB=&quot;/home/student/lab&quot;
export CODE=$LAB/code
export DATA=$PRE/data
export MODEL=$PRE/models
export RESULTS=$PRE/results
export METAXCAN=$PRE/repos/MetaXcan-master/software</code></pre>
<div id="predict-expression" class="section level2">
<h2>predict expression</h2>
<div class="figure">
<img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/PrediXcan-run.png" alt="" />
<p class="caption">Visual summary of predixcan runs</p>
</div>
<ul>
<li><p>Remember you need to copy and paste this code chunk into the terminal to run it. Also make sure you activated the imlabtools environment which has all the necessary python modules.</p></li>
<li><p>Make sure all the paths and file names are correct. This run should take about one minute.</p></li>
</ul>
<pre class="bash"><code>
printf &quot;Predict expression\n\n&quot;
python3 $METAXCAN/Predict.py \
--model_db_path $PRE/models/gtex_v8_en/en_Whole_Blood.db \
--vcf_genotypes $DATA/predixcan/genotype/filtered.vcf.gz \
--vcf_mode genotyped \
--variant_mapping $DATA/predixcan/gtex_v8_eur_filtered_maf0.01_monoallelic_variants.txt.gz id rsid \
--on_the_fly_mapping METADATA &quot;chr{}_{}_{}_{}_b38&quot; \
--prediction_output $RESULTS/predixcan/Whole_Blood__predict.txt \
--prediction_summary_output $RESULTS/predixcan/Whole_Blood__summary.txt \
--verbosity 9 \
--throw
</code></pre>
</div>
<div id="check-predicted-values" class="section level2">
<h2>check predicted values</h2>
<pre class="r"><code>prediction_fp = glue::glue(&quot;{RESULTS}/predixcan/Whole_Blood__predict.txt&quot;)
## Read the Predict.py output into a dataframe
predicted_expression = read.table(file=prediction_fp, sep=&quot;\t&quot;, quote=&quot;&quot;, comment.char=&quot;&quot;, skip = 1, header = TRUE)
# Retain the column names
cols = read.table(file=prediction_fp, sep=&quot;\t&quot;, quote=&quot;&quot;, comment.char=&quot;&quot;, nrows = 1)
# Fill the column names
colnames(predicted_expression) = unname(unlist(cols[1,]))
## Melt the data so each row is FID, IID, gene_id, predicted_expression
predicted_expression = predicted_expression %&gt;%
  pivot_longer(
    cols = starts_with(&quot;ENSG&quot;),
    names_to = &quot;gene_id&quot;,
    values_to = &quot;predicted_expression&quot;,
    values_drop_na = TRUE
  )
## Remove the decimal points from the gene_id&#39;s
predicted_expression$gene_id = gsub(&quot;\\..*&quot;,&quot;&quot;,predicted_expression$gene_id)
head(predicted_expression)

## read summary of prediction, number of SNPs per gene, cross validated prediction performance
prediction_summary = read_tsv(glue::glue(&quot;{RESULTS}/predixcan/Whole_Blood__summary.txt&quot;))
## number of genes with a prediction model
dim(prediction_summary)
head(prediction_summary)

print(&quot;distribution of prediction performance r2&quot;)
summary(prediction_summary$pred_perf_r2)</code></pre>
</div>
<div id="assess-prediction-performance-optional" class="section level2">
<h2>assess prediction performance (optional)</h2>
<pre class="r"><code>## merge with GEUVADIS expression data

## calculate Spearman correlation

## select a few genes and plot predicted vs observed expression</code></pre>
</div>
<div id="run-association-with-a-phenotype" class="section level2">
<h2>run association with a phenotype</h2>
<pre class="bash"><code>
export PHENO=&quot;sim.infinitesimal_pve0.1&quot;

printf &quot;association\n\n&quot;
python3 $METAXCAN/PrediXcanAssociation.py \
--expression_file $RESULTS/predixcan/Whole_Blood__predict.txt \
--input_phenos_file $DATA/predixcan/phenotype/$PHENO.txt \
--input_phenos_column pheno \
--output $RESULTS/predixcan/$PHENO/Whole_Blood__association.txt \
--verbosity 9 \
--throw

export PHENO=&quot;sim.spike_n_slab_0.01_pve0.1&quot;

printf &quot;association\n\n&quot;
python3 $METAXCAN/PrediXcanAssociation.py \
--expression_file $RESULTS/predixcan/Whole_Blood__predict.txt \
--input_phenos_file $DATA/predixcan/phenotype/$PHENO.txt \
--input_phenos_column pheno \
--output $RESULTS/predixcan/$PHENO/Whole_Blood__association.txt \
--verbosity 9 \
--throw

export PHENO=&quot;sim.spike_n_slab_0.1_pve0.05&quot;

printf &quot;association\n\n&quot;
python3 $METAXCAN/PrediXcanAssociation.py \
--expression_file $RESULTS/predixcan/Whole_Blood__predict.txt \
--input_phenos_file $DATA/predixcan/phenotype/$PHENO.txt \
--input_phenos_column pheno \
--output $RESULTS/predixcan/$PHENO/Whole_Blood__association.txt \
--verbosity 9 \
--throw
</code></pre>
</div>
<div id="read-association-results" class="section level2">
<h2>read association results</h2>
<pre class="r"><code>## read association results
PHENO=&quot;sim.spike_n_slab_0.01_pve0.1&quot;


predixcan_association = read_tsv(glue::glue(&quot;{RESULTS}/predixcan/{PHENO}/Whole_Blood__association.txt&quot;))

## take a look at the results
dim(predixcan_association)
predixcan_association %&gt;% arrange(pvalue) %&gt;% head
predixcan_association %&gt;% arrange(pvalue) %&gt;% ggplot(aes(pvalue)) + geom_histogram(bins=20)
## compare distribution against the null (uniform)
gg_qqplot(predixcan_association$pvalue, max_yval = 40)


truebetas = read_tsv(glue::glue(&quot;{DATA}/predixcan/phenotype/gene-effects/{PHENO}.txt&quot;))
truebetas = (predixcan_association %&gt;% 
               inner_join(truebetas,by=c(&quot;gene&quot;=&quot;gene_id&quot;)) %&gt;%
               select(c(&#39;predicted_beta&#39;=&#39;effect&#39;, &#39;true_beta&#39;=&#39;effect_size&#39;,&#39;pvalue&#39;)))

truebetas %&gt;% ggplot(aes(predicted_beta, true_beta))+geom_point()+geom_abline()</code></pre>
</div>
<div id="exercise" class="section level2">
<h2>Exercise</h2>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
show top genes</li>
<li><input type="checkbox" disabled="" />
compare with true effect sizes</li>
<li><input type="checkbox" disabled="" />
interpret</li>
</ul>
<pre class="r"><code># truebetas %&gt;% head()

#suppressPackageStartupMessages(library(qqman))</code></pre>
</div>
<div id="exercise-1" class="section level2">
<h2>Exercise</h2>
<pre class="r"><code>## get chr and position of genes (transcription start site)

## do you see examples of LD contamination?</code></pre>
</div>
<div id="section" class="section level2">
<h2>——-</h2>
</div>
</div>
<div id="summary-predixcan" class="section level1">
<h1>Summary PrediXcan</h1>
<p>Now we will use the summary results from a GWAS of coronary artery disease to calculate the association between the genetic component of the expression of genes and coronary artery disease risk. We will use the SPrediXcan.py.</p>
<div class="figure">
<img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/gwas-PrediXcan-spredixcan.png" alt="" />
<p class="caption">Visual summary of s-predixcan</p>
</div>
<pre class="r"><code>## harmonized and imputed GWAS result for coronary artery disease is available in 
# $PRE/spredixcan/data/</code></pre>
<div id="run-s-predixcan" class="section level2">
<h2>run s-predixcan</h2>
<pre class="bash"><code>
export PRE=&quot;/home/student/QGT-Columbia-HKI&quot;
export DATA=$PRE/data
export MODEL=$PRE/models
export METAXCAN=$PRE/repos/MetaXcan-master/software
export RESULTS=$PRE/results

python $METAXCAN/SPrediXcan.py \
--gwas_file  $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
--snp_column panel_variant_id --effect_allele_column effect_allele --non_effect_allele_column non_effect_allele --zscore_column zscore \
--model_db_path $MODEL/gtex_v8_mashr/mashr_Whole_Blood.db \
--covariance $MODEL/gtex_v8_mashr/mashr_Whole_Blood.txt.gz \
--keep_non_rsid --additional_output --model_db_snp_key varID \
--throw \
--output_file $RESULTS/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv
</code></pre>
</div>
<div id="plot-and-interpret-s-predixcan-results" class="section level2">
<h2>plot and interpret s-predixcan results</h2>
<pre class="r"><code>spredixcan_association = read_csv(glue::glue(&quot;{RESULTS}/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv&quot;))
dim(spredixcan_association)
spredixcan_association %&gt;% arrange(pvalue) %&gt;% head
spredixcan_association %&gt;% arrange(pvalue) %&gt;% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association$pvalue)</code></pre>
<ul class="task-list">
<li><input type="checkbox" disabled="" />
SORT1, considered to be a causal gene for LDL cholesterol and as a consequence of coronary artery disease, is not found here. Why? (tissue)</li>
</ul>
</div>
<div id="exercise-2" class="section level2">
<h2>Exercise</h2>
<ul class="task-list">
<li><p><input type="checkbox" disabled="" />
run s-predixcan with liver model, do you find SORT1? Is it significant?</p></li>
<li><p><input type="checkbox" disabled="" />
compare zscores in liver and whole blood.</p></li>
</ul>
</div>
<div id="run-multixcan-optional" class="section level2">
<h2>run multixcan (optional)</h2>
<pre class="bash"><code>
export MODEL=$PRE/models
export DATA=$PRE/data

python $METAXCAN/SMulTiXcan.py \
--models_folder $MODEL/gtex_v8_mashr \
--models_name_pattern &quot;mashr_(.*).db&quot; \
--snp_covariance $MODEL/gtex_v8_expression_mashr_snp_smultixcan_covariance.txt.gz \
--metaxcan_folder $RESULTS/spredixcan/eqtl/ \
--metaxcan_filter &quot;CARDIoGRAM_C4D_CAD_ADDITIVE__PM__(.*).csv&quot; \
--metaxcan_file_name_parse_pattern &quot;(.*)__PM__(.*).csv&quot; \
--gwas_file $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
--snp_column panel_variant_id --effect_allele_column effect_allele --non_effect_allele_column non_effect_allele --zscore_column zscore --keep_non_rsid --model_db_snp_key varID \
--cutoff_condition_number 30 \
--verbosity 7 \
--throw \
--output $RESULTS/smultixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE_smultixcan.txt
</code></pre>
</div>
</div>
<div id="colocalization-methods" class="section level1">
<h1>Colocalization methods</h1>
<ul>
<li>Colocalization methods seek to estimate the probability that the complex trait and expression causal variants are the same. We favor methods that calculate the probability of causality for each trait (posterior inclusion probability), called fine-mapping methods. Here we use torus for fine-mapping and fastENLOC for colocalization.</li>
</ul>
<div class="figure">
<img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/colocalization-run.png" alt="" />
<p class="caption">Visual summary of colocalization</p>
</div>
<div id="gwas-summary-statistics-to-torus-format" class="section level2">
<h2>GWAS summary statistics to torus format</h2>
<pre class="bash"><code>
##TODO CAD GWAS is in hg38

python $CODE/gwas_to_torus_zscore.py \
-input_gwas $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
-input_ld_regions $DATA/spredixcan/eur_ld_hg38.txt.gz \
-output_fp $DATA/fastenloc/CARDIoGRAM_C4D_CAD_ADDITIVE.zval.gz
</code></pre>
</div>
<div id="fine-map-gwas-results" class="section level2">
<h2>fine-map GWAS results</h2>
<ul>
<li><p>We will run torus due to time limitation but ideally we would like to run a method that allows multiple causal variants per locus, such as DAP-G or SusieR.</p></li>
<li><p>torus has been precompiled and placed within the PATH</p></li>
</ul>
<pre class="bash"><code>
export TORUSOFT=torus

$TORUSOFT -d $PRE/data/fastenloc/CARDIoGRAM_C4D_CAD_ADDITIVE.zval.gz --load_zval -dump_pip $PRE/data/fastenloc/CARDIoGRAM_C4D_CAD_ADDITIVE.pip
cd $PRE/data/fastenloc
gzip CAD.gwas.pip
cd $PRE 
</code></pre>
<p>We can take a quick look at the z-values and finemapping PIPs:</p>
<p>The inputs have columns SNP_ID, LOCUS_ID, ZVAL, and the outputs have columns SNP_ID, LOCUS_ID, PVAL, PIP (???)</p>
<pre class="bash"><code>cd $PRE/data/fastenloc
zless Height.torus.zval.gz

zless Height.gwas.pip</code></pre>
</div>
<div id="calculate-colocalization-with-fastenloc" class="section level2">
<h2>calculate colocalization with fastENLOC</h2>
<pre class="bash"><code>## check out tutorial https://github.com/xqwen/fastenloc/tree/master/tutorial

export eqtl_annotation_gzipped=$PRE/data/fastenloc/FASTENLOC-gtex_v8.eqtl_annot.vcf.gz
export gwas_data_gzipped=$PRE/data/fastenloc/CARDIoGRAM_C4D_CAD_ADDITIVE.gwas.pip.gz
export TISSUE=Whole_Blood
export FASTENLOCSOFT=fastenloc
export FASTENLOCSOFT=/Users/owenmelia/projects/finemapping_bin/src/fastenloc/src/fastenloc

mkdir $RESULTS/fastenloc/
cd $RESULTS/fastenloc/
$FASTENLOCSOFT -eqtl $eqtl_annotation_gzipped -gwas $gwas_data_gzipped -t $TISSUE 

#[-total_variants total_snp] [-thread n] [-prefix prefix_name] [-s shrinkage]
</code></pre>
</div>
<div id="analyze-results" class="section level2">
<h2>analyze results</h2>
<pre class="r"><code>## optional - compare with s-predixcan results

fastenloc_results = load_fastenloc_coloc_result(glue::glue(&quot;{RESULTS}/fastenloc/enloc.sig.out&quot;))

spredixcan_and_fastenloc = inner_join(spredixcan_association, fastenloc_results, by=c(&#39;gene&#39;=&#39;Signal&#39;))

ggplot(spredixcan_and_fastenloc, aes(RCP, -log10(pvalue))) + geom_point()</code></pre>
<hr />
</div>
</div>
<div id="mendelian-randomization-methods" class="section level1">
<h1>Mendelian randomization methods</h1>
<div id="run-twmr-for-a-locus" class="section level2">
<h2>run TWMR (for a locus)</h2>
<div class="figure">
<img src="https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/TWMR.png" alt="" />
<p class="caption">TWMR</p>
</div>
<pre class="bash"><code>
export TWMR=$PRE/repos/TWMR-master
export OUTPUT=$PRE/results
GENE=ENSG00000002919

cd $TWMR

R &lt; $TWMR/MR.R --no-save $GENE

cd $PRE

## output: /home/student/QGT-Columbia-HKI/repos/TWMR-master/ENSG00000002919.alpha
</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.0   stringr_1.4.0   dplyr_0.8.5     purrr_0.3.4    
[5] readr_1.3.1     tidyr_1.0.2     tibble_3.0.0    ggplot2_3.3.0  
[9] tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.0.0 xfun_0.13        haven_2.2.0      lattice_0.20-41 
 [5] colorspace_1.4-1 vctrs_0.2.4      generics_0.0.2   htmltools_0.4.0 
 [9] yaml_2.2.1       rlang_0.4.5      later_1.0.0      pillar_1.4.3    
[13] withr_2.1.2      glue_1.4.1       DBI_1.1.0        dbplyr_1.4.2    
[17] readxl_1.3.1     modelr_0.1.6     lifecycle_0.2.0  cellranger_1.1.0
[21] munsell_0.5.0    gtable_0.3.0     workflowr_1.6.2  rvest_0.3.5     
[25] evaluate_0.14    knitr_1.28       httpuv_1.5.2     fansi_0.4.1     
[29] broom_0.5.5      Rcpp_1.0.4.6     promises_1.1.0   backports_1.1.6 
[33] scales_1.1.0     jsonlite_1.6.1   fs_1.4.1         hms_0.5.3       
[37] digest_0.6.25    stringi_1.4.6    rprojroot_1.3-2  grid_3.6.3      
[41] cli_2.0.2        tools_3.6.3      magrittr_1.5     crayon_1.3.4    
[45] whisker_0.4      pkgconfig_2.0.3  ellipsis_0.3.0   xml2_1.3.1      
[49] reprex_0.3.0     lubridate_1.7.8  rstudioapi_0.11  assertthat_0.2.1
[53] rmarkdown_2.1    httr_1.4.1       R6_2.4.1         nlme_3.1-147    
[57] git2r_0.26.1     compiler_3.6.3  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
