---
title: "QGT-Columbia-analysis-plan"
author: "Hae Kyung Im"
date: "2020-06-03"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Set up 

- [ ] download data and software [from Box](https://uchicago.box.com/s/zhapf2zfxcpj7thvq4sjnqale3emleum).
  This will have copies of all the software repositories and the models

Linux is the operating system of choice to run bioinformatics software. Here are offering two options

- Option 1: full setup, recommended for the linux-savvy with full setup
- Option 2: pre-installed RStudio in Google cloud, recommended for people less familiar with linux

The latest version of the analysis plan markdown document that generated this page is on [github here](https://github.com/hakyimlab/QGT-Columbia-HKI/blob/master/analysis/analysis_plan.Rmd)
rendered [here as an html page](https://hakyimlab.github.io/QGT-Columbia-HKI/analysis_plan.html)

# Option 1

- [ ] install anaconda/miniconda
- [ ] define imlabtools conda environment [how to here](https://github.com/hakyimlab/MetaXcan/blob/master/README.md#example-conda-environment-setup), which will install all the python modules needed for this analysis session
- [x] download software (copies of the repos are already included in the course folder QCT-Columbia-HKI/repos/)
  - download metaxcan repo
  - download torus repo
  - download fastenloc repo
  - download TMWR repo
- [x] download prediction models from predictdb.org (a few models are included in the course folder QCT-Columbia-HKI/repos/)
- [ ] install R/RStudio/tidyverse package
- [ ] (optional) install workflowr package in R
- [ ] git clone https://github.com/hakyimlab/QGT-Columbia-HKI.git
- [ ] start Rstudio (if you installed workflowr, you can just open the QGT-Columbia-HKI.Rproj)

# Option 2
- [ ] claim your Rstudio server IP address ()
- [ ] connect to the Rstudio server using the url you claimed (http://xxx.xxx.xxx.xxx:8787)

# Both options

- [ ] update the analysis document

```{bash eval=FALSE}
PRE="/home/student/"
cd $PRE/lab/
git pull 
```

- [ ] activate the the imlabtools environment
```{bash, eval=FALSE}
conda activate imlabtools
```

** Notice that the bash chunks need to be copy-pasted to the terminal, not performed within the chunk.

## Summary of analysis plan

- predict whole blood expression
- check how well the prediction works with GEUVADIS expression data
- run association between predicted expression and a simulated phenotype
- calculate association between expression levels and coronary artery disease risk using s-predixcan
- fine-map the coronary artery disease gwas results using torus
- calculate colocalization probability using fastenloc
- run transcriptome-wide mendelian randomization in one locus of interest


```{r preliminary definitions}

suppressPackageStartupMessages(library(tidyverse))

```

# Transcriptome-wide association methods

- [ ] define some variables to access the data more easily within the R session. Run the following r chunk


```{r preliminaries, eval=FALSE}

print(getwd())

lab="/home/student/lab"
CODE=glue::glue("{lab}/code")
source(glue::glue("{CODE}/extra_functions.R"))
#source(glue::glue("code/extra_functions.R"))

PRE="/home/student/QGT-Columbia-HKI"
MODEL=glue::glue("{PRE}/models")
DATA=glue::glue("{PRE}/data")
RESULTS=glue::glue("{PRE}/results")
METAXCAN=glue::glue("{PRE}/repos/MetaXcan-master/software")
FASTENLOC=glue::glue("{PRE}/repos/fastenloc-master")
TORUS=glue::glue("{PRE}/repos/torus-master")
TWMR=glue::glue("{PRE}/repos/TWMR-master")
```

- [ ] define some variables to access the data more easily in the terminal. Run the following bash chunk. You will need to copy and paste the following chunk in the terminal


```{bash folder name variables, eval=FALSE}
export PRE="/home/student/QGT-Columbia-HKI"
export LAB="/home/student/lab"
export CODE=$LAB/code
export DATA=$PRE/data
export MODEL=$PRE/models
export RESULTS=$PRE/results
export METAXCAN=$PRE/repos/MetaXcan-master/software
```

## predict expression 

![Visual summary of predixcan runs](https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/PrediXcan-run.png)

- Remember you need to copy and paste this code chunk into the terminal to run it. Also make sure you activated the imlabtools environment which has all the necessary python modules. 

- Make sure all the paths and file names are correct. This run should take about one minute.

```{bash predict genetic component of expression, eval=FALSE}

printf "Predict expression\n\n"
python3 $METAXCAN/Predict.py \
--model_db_path $PRE/models/gtex_v8_en/en_Whole_Blood.db \
--vcf_genotypes $DATA/predixcan/genotype/filtered.vcf.gz \
--vcf_mode genotyped \
--variant_mapping $DATA/predixcan/gtex_v8_eur_filtered_maf0.01_monoallelic_variants.txt.gz id rsid \
--on_the_fly_mapping METADATA "chr{}_{}_{}_{}_b38" \
--prediction_output $RESULTS/predixcan/Whole_Blood__predict.txt \
--prediction_summary_output $RESULTS/predixcan/Whole_Blood__summary.txt \
--verbosity 9 \
--throw

```

## check predicted values

```{r check prediction performance, eval=FALSE}

## read matrix of predicted expression
predicted_expression = read_tsv(glue::glue("{RESULTS}/predixcan/Whole_Blood__predict.txt"))
dim(predicted_expression)
head(predicted_expression[,1:5])

## read summary of prediction, number of SNPs per gene, cross validated prediction performance
prediction_summary = read_tsv(glue::glue("{RESULTS}/predixcan/Whole_Blood__summary.txt"))
## number of genes with a prediction model
dim(prediction_summary)
head(prediction_summary)

print("distribution of prediction performance r2")
summary(prediction_summary$pred_perf_r2)

```

## assess prediction performance (optional)

```{r assess prediction performance, eval=FALSE}
## merge with GEUVADIS expression data

## calculate Spearman correlation

## select a few genes and plot predicted vs observed expression

```

## run association with a phenotype

```{bash run predixcan association, eval=FALSE}

export PHENO="sim.infinitesimal_pve0.1"

printf "association\n\n"
python3 $METAXCAN/PrediXcanAssociation.py \
--expression_file $RESULTS/predixcan/Whole_Blood__predict.txt \
--input_phenos_file $DATA/predixcan/phenotype/$PHENO.txt \
--input_phenos_column pheno \
--output $RESULTS/predixcan/$PHENO/Whole_Blood__association.txt \
--verbosity 9 \
--throw

export PHENO="sim.spike_n_slab_0.01_pve0.1"

printf "association\n\n"
python3 $METAXCAN/PrediXcanAssociation.py \
--expression_file $RESULTS/predixcan/Whole_Blood__predict.txt \
--input_phenos_file $DATA/predixcan/phenotype/$PHENO.txt \
--input_phenos_column pheno \
--output $RESULTS/predixcan/$PHENO/Whole_Blood__association.txt \
--verbosity 9 \
--throw

export PHENO="sim.spike_n_slab_0.1_pve0.05"

printf "association\n\n"
python3 $METAXCAN/PrediXcanAssociation.py \
--expression_file $RESULTS/predixcan/Whole_Blood__predict.txt \
--input_phenos_file $DATA/predixcan/phenotype/$PHENO.txt \
--input_phenos_column pheno \
--output $RESULTS/predixcan/$PHENO/Whole_Blood__association.txt \
--verbosity 9 \
--throw

```

## read association results

```{r read predixcan association results, eval=FALSE}

## read association results
PHENO="sim.spike_n_slab_0.01_pve0.1"


predixcan_association = read_tsv(glue::glue("{RESULTS}/predixcan/{PHENO}/Whole_Blood__association.txt"))

## take a look at the results
dim(predixcan_association)
predixcan_association %>% arrange(pvalue) %>% head
predixcan_association %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)
## compare distribution against the null (uniform)
gg_qqplot(predixcan_association$pvalue, max_yval = 40)


truebetas = read_tsv(glue::glue("{DATA}/predixcan/phenotype/gene-effects/{PHENO}.txt"))
truebetas = (predixcan_association %>% 
               inner_join(truebetas,by=c("gene"="gene_id")) %>%
               select(c('predicted_beta'='effect', 'true_beta'='effect_size','pvalue')))

truebetas %>% ggplot(aes(predicted_beta, true_beta))+geom_point()+geom_abline()
```

## Exercise
- [ ] show top genes
- [ ] compare with true effect sizes
- [ ] interpret

```{r}
# truebetas %>% head()

#suppressPackageStartupMessages(library(qqman))

```

## Exercise

```{r}

## get chr and position of genes (transcription start site)

## do you see examples of LD contamination?

```



-------
-------

# Summary PrediXcan

Now we will use the summary results from a GWAS of coronary artery disease to calculate the association between the genetic component of the expression of genes and coronary artery disease risk. We will use the SPrediXcan.py.

![Visual summary of s-predixcan](https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/gwas-PrediXcan-spredixcan.png)


```{r}

## harmonized and imputed GWAS result for coronary artery disease is available in 
# $PRE/spredixcan/data/

```

## run s-predixcan 

```{bash run s-predixcan, eval=FALSE}

export PRE="/home/student/QGT-Columbia-HKI"
export DATA=$PRE/data
export MODEL=$PRE/models
export METAXCAN=$PRE/repos/MetaXcan-master/software
export RESULTS=$PRE/results

python $METAXCAN/SPrediXcan.py \
--gwas_file  $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
--snp_column panel_variant_id --effect_allele_column effect_allele --non_effect_allele_column non_effect_allele --zscore_column zscore \
--model_db_path $MODEL/gtex_v8_mashr/mashr_Whole_Blood.db \
--covariance $MODEL/gtex_v8_mashr/mashr_Whole_Blood.txt.gz \
--keep_non_rsid --additional_output --model_db_snp_key varID \
--throw \
--output_file $RESULTS/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv

```

## plot and interpret s-predixcan results

```{r analyze s-predixcan results, eval=FALSE}

spredixcan_association = read_csv(glue::glue("{RESULTS}/spredixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE__PM__Whole_Blood.csv"))
dim(spredixcan_association)
spredixcan_association %>% arrange(pvalue) %>% head
spredixcan_association %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association$pvalue)

```

- [ ] SORT1, considered to be a causal gene for LDL cholesterol and as a consequence of coronary artery disease, is not found here. Why? (tissue)

## Exercise

- [ ] run s-predixcan with liver model, do you find SORT1? Is it significant?

- [ ] compare zscores in liver and whole blood.



## run multixcan (optional)

```{bash run multixcan, eval=FALSE}

export MODEL=$PRE/models
export DATA=$PRE/data

python $METAXCAN/SMulTiXcan.py \
--models_folder $MODEL/gtex_v8_mashr \
--models_name_pattern "mashr_(.*).db" \
--snp_covariance $MODEL/gtex_v8_expression_mashr_snp_smultixcan_covariance.txt.gz \
--metaxcan_folder $RESULTS/spredixcan/eqtl/ \
--metaxcan_filter "CARDIoGRAM_C4D_CAD_ADDITIVE__PM__(.*).csv" \
--metaxcan_file_name_parse_pattern "(.*)__PM__(.*).csv" \
--gwas_file $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
--snp_column panel_variant_id --effect_allele_column effect_allele --non_effect_allele_column non_effect_allele --zscore_column zscore --keep_non_rsid --model_db_snp_key varID \
--cutoff_condition_number 30 \
--verbosity 7 \
--throw \
--output $RESULTS/smultixcan/eqtl/CARDIoGRAM_C4D_CAD_ADDITIVE_smultixcan.txt

```

# Colocalization methods

- Colocalization methods seek to estimate the probability that the complex trait and expression causal variants are the same. We favor methods that calculate the probability of causality for each trait (posterior inclusion probability), called fine-mapping methods. Here we use torus for fine-mapping and fastENLOC for colocalization.

![Visual summary of colocalization](https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/colocalization-run.png)

## GWAS summary statistics to torus format

```{bash, eval=FALSE}

##TODO CAD GWAS is in hg38

python $CODE/gwas_to_torus_zscore.py \
-input_gwas $DATA/spredixcan/imputed_CARDIoGRAM_C4D_CAD_ADDITIVE.txt.gz \
-input_ld_regions $DATA/spredixcan/eur_ld_hg38.txt.gz \
-output_fp $DATA/fastenloc/CARDIoGRAM_C4D_CAD_ADDITIVE.zval.gz

```

## fine-map GWAS results

- We will run torus due to time limitation but ideally we would like to run a method that allows multiple causal variants per locus, such as DAP-G or SusieR.

- torus has been precompiled and placed within the PATH


```{bash run torus, eval=FALSE}

export TORUSOFT=torus

$TORUSOFT -d $PRE/data/fastenloc/CARDIoGRAM_C4D_CAD_ADDITIVE.zval.gz --load_zval -dump_pip $PRE/data/fastenloc/Height.gwas.pip
cd $PRE/data/fastenloc
gzip Height.gwas.pip
cd $PRE 

```

We can take a quick look at the z-values and finemapping PIPs:

The inputs have columns SNP_ID, LOCUS_ID, ZVAL, and the outputs have columns SNP_ID, LOCUS_ID, PVAL, PIP (???)
```{bash, eval=FALSE}
cd $PRE/data/fastenloc
zless Height.torus.zval.gz

zless Height.gwas.pip
```



## calculate colocalization with fastENLOC

```{bash run fastENLOC, eval=FALSE}
## check out tutorial https://github.com/xqwen/fastenloc/tree/master/tutorial

export eqtl_annotation_gzipped=$PRE/data/fastenloc/FASTENLOC-gtex_v8.eqtl_annot.vcf.gz
export gwas_data_gzipped=$PRE/data/fastenloc/Height.gwas.pip.gz
export TISSUE=Whole_Blood
export FASTENLOCSOFT=fastenloc
export FASTENLOCSOFT=/Users/owenmelia/projects/finemapping_bin/src/fastenloc/src/fastenloc

mkdir $RESULTS/fastenloc/
cd $RESULTS/fastenloc/
$FASTENLOCSOFT -eqtl $eqtl_annotation_gzipped -gwas $gwas_data_gzipped -t $TISSUE 

#[-total_variants total_snp] [-thread n] [-prefix prefix_name] [-s shrinkage]

```

## analyze results 

```{r analyze torus results, eval=FALSE}

## optional - compare with s-predixcan results

fastenloc_results = load_fastenloc_coloc_result(glue::glue("{RESULTS}/fastenloc/enloc.sig.out"))

spredixcan_and_fastenloc = inner_join(spredixcan_association, fastenloc_results, by=c('gene'='Signal'))

ggplot(spredixcan_and_fastenloc, aes(RCP, -log10(pvalue))) + geom_point()

```

----------
# Mendelian randomization methods

## run  TWMR (for a locus)
![TWMR](https://raw.githubusercontent.com/hakyimlab/QGT-Columbia-HKI/master/extras/figures/TWMR.png)

```{bash run TWMR, eval=FALSE}

export TWMR=$PRE/repos/TWMR-master
export OUTPUT=$PRE/results
GENE=ENSG00000002919

cd $TWMR

R < $TWMR/MR.R --no-save $GENE

cd $PRE

## output: /home/student/QGT-Columbia-HKI/repos/TWMR-master/ENSG00000002919.alpha

```

```{r analyze TWMR results, eval=FALSE}

```

